### ISSUE TITLE
Iris data set training

### ISSUE THREAD

**Issue by AyronK** [AyronK/softmax-dnn-example#1]
This issue is focused mainly on getting knowledge about neural networks with Accord.Net. All conclusions and ideas will be posted in comments and quoted in appropriate place.

# Step 1

As first step I decided to create a simple example based on teaching `Accord.Neuro.ActivationNetwork` with famous Iris data set using `BackPropagationLearning`. The goal of this issue is to get some basic experience with Accord.Net freamwork.

- [x] Shuffling data set
- [x] Dividing set to learning and testing set
- [x] Learning ActivationNetwork
- [x] Calculating accuracy for both learning and testing set
- [x] Using softmax function to simulate network prediction process

# Step 2

Next I'm refactoring code for better customization and try different tools in Accord.Net. At this point I also try to predict what modifications I'll need in my future work and check if it is possible with this framework.

- [x] Cleanup iris example
- [x] Check out `Accord.Neuro.Networks.DeepBeliefNetwork`
- [x] ~Check out `Accord.Neuro.Networks.RestrictedBoltzmannMachine`~ 
_Limited to one hidden layer - leaving it for now_
- [x] Check out different learning techniques

### Conclusions

# Step 3

I am now testing and tuning different parameters for both networks and learning alghorithms. Also I'm gathering further reasearch on deep belief networks and unsupervised learning to better understand the topic.

> One thing I might enhance is extending a default network with additional build in softmax layer which should not be a problem.

> Learning alghorithms, though, might not be enough if I had a need to include softmax output comparision in error calculation. Than I may need to implement my own algorithms.

**Comment by AyronK** [comment]
Using unsupervised learning to teach DBN makes training much faster on multiple layers with many hidden neurons. What I've read though, it might be less effective if training data set is far from perfect.

**Comment by AyronK** [comment]
I'm closing issue once I've tried some options available with `Accord.Net` on Iris dataset. Certainly, more research is necessary but now I'll focus on data operations.
